pkg/runtime/runtime.go

```diff
@@ -842,17 +861,28 @@ func (a *DaprRuntime) getTopicRoutes() map[string]string {
 	}
 
 	for _, s := range subscriptions {
-		topicRoutes[s.Topic] = s.Route
+		if _, ok := a.pubSubs[s.PubsubName]; !ok {
+			msg := fmt.Sprintf("cannot subscribe to pubsub %s - does not exist", s.PubsubName)
+			return nil, errors.New(msg)
```

Comment:
we should log and ignore the subscription for which there was no pubsub system, and continue processing rest of subscription

Reply:
Is that what's done for other components?  It seems this situation can be hit in 2 cases:
1. misconfiguration of either the cluster the app (e.g. a pubsub component the app refers to is not there)
2. some odd transient error.

In the case of item 1 I don't think the app should run, and so it should just fail, which is what this edit does.  

In the case of item 2, with this code the app will restart and if the transient error is gone then it will function normally.  If the transient error happens but daprd ignores failed components, it ends up running in some "half initialized" mode which seems incorrect and confusing to the user.

Reply:
I agree that we should log errors and continue.

If the user code is not receiving events, the logs will be checked and the error will surface up, especially in a production environment when errors are forwarded to a backed like Elastic this is visible. However the Dapr sidecar should continue to operate even in the face of errors, as it may be handling other parts of the system and should not cease to operate completely because a component has been misconfigured.

Reply:
Ok, made change.